moshi/src/main/java/com/squareup/moshi/ArrayJsonAdapter.java

```diff
@@ -69,4 +69,8 @@
     }
     writer.endArray();
   }
+
+  @Override public String toString() {
+    return "JsonAdapter(" + elementClass.getSimpleName() + ").array()";
```

Comment:
makes sense, but then we can get something like `JsonAdapter(Integer).nullSafe()[]`. I'm between that and `elementaAdapter + ".array()"`, in both cases we maintain the chain.